### PR TITLE
Harden TLS

### PR DESCRIPTION
**Pull request by phrag**
Strongest Ciphers / EECDH

### CODE REVIEW

**Comment by M5oul** [comment]
I noticed that [Calomel extension](https://addons.mozilla.org/en-US/firefox/addon/calomel-ssl-validation) now say that SHA-1 is not secure with 4/15 as note.

**Comment by M5oul** [comment]
I tried it.
It did not changed the SHA issue.

File: data/templates/nginx/plain/yunohost_admin.conf [hunk 1]
```diff
@@ -21,9 +21,13 @@ server {
...
     ssl_prefer_server_ciphers on;
     ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
-    ssl_ciphers ALL:!aNULL:!eNULL:!LOW:!EXP:!RC4:!3DES:+HIGH:+MEDIUM;
+    ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+    ssl_ecdh_curve secp384r1;
+    ssl_stapling on;
+    ssl_stapling_verify on;
 
     add_header Strict-Transport-Security "max-age=31536000;";
+    add_header X-Content-Type-Options nosniff;
```
Comment (julienmalik) [inline data/templates/nginx/plain/yunohost_admin.conf]: didn't know about this one, but seems relevant : http://stackoverflow.com/questions/18337630/what-is-x-content-type-options-nosniff

**Comment by julienmalik** [comment]
You only modified yunohost_admin.conf, but the server conf (server.tpl) also does embed ssl directives.

Best would probably be to factorize generic ssl options (all ssl_xxx except the certs) in a common file, to be used in domain specific conf and yunohost_admin.conf

**Comment by julienmalik** [comment]
What recommendation did you follow to set up these new cipher list and the rest ?

I think we used the Mozilla SSL conf generator : https://mozilla.github.io/server-side-tls/ssl-config-generator/
which is good since it quickly shows the corresponding supported clients.

The choice to harden ssl configuration usually has consequences on the number of supported clients (browsers), and on performance (client or server side). Ideally that's something that should be chosen by the admin, but yunohost is not yet ready for this kind of configuration.

**Comment by zamentur** [comment]
Typically, it's here that we should think to introduce a settings management
In this case it could be:

- Harden encryption (some old device couldn't reach this server)
- Classic encryption

**Comment by Psycojoker** [comment]
This one seems to be depending on https://github.com/YunoHost/yunohost/pull/229

**Comment by alexAubin** [comment]
Closing because it seems to be superseded by https://github.com/YunoHost/yunohost/pull/259

**Pull request closed without merging**